bundler/lib/bundler/lockfile_parser.rb

```diff
@@ -149,11 +160,11 @@ def parse_dependency(line)
       return unless line =~ NAME_VERSION
       spaces = $1
       return unless spaces.size == 2
-      name = $2
+      name = -$2
```

Comment:
Cute. I'm surprised this isn't used more often, enough that it's common practice. First I've seen it and had to look it up.

Reply:
I don't think I've seen this before, but I failed when I tried to look it up. What is it?

Reply:
Unary minus on String. It freezes the string. Unary + thaws.

Reply:
Huh, how about that. I guess I should have checked the API docs! I did finally find a writeup: https://metaredux.com/posts/2019/05/10/weird-ruby-positive-and-negative-strings.html

Reply:
`@-` is an alias for `#dedup`, which not only returns a frozen string, but will deduplicate it, leading to reduced retained memory usage

Reply:
FYI: The unary operator is `-@`.